src/main/java/com/revature/controllers/PersonController.java

```diff
@@ -24,99 +27,102 @@
+
+			Person person = personLogic.getPersonById(id);
 		
+			
+			if (person != null) {
+				
+				return ResponseEntity.ok(person);
+				
+			} else {
+				
+				System.out.println("ERROR!");				
+				return new ResponseEntity<>(HttpStatus.NOT_FOUND);
+			}
+			
+			
+	}
+		
+	
 	@RequestMapping(method = RequestMethod.GET, value = "persons")
-	public ResponseEntity<List<Person>> getPerson(@RequestParam(defaultValue="", required=false) String firstname,
-			@RequestParam(defaultValue="", required=false) String lastname, 
-			@RequestParam(defaultValue="0", required=false) Integer id,
+	public ResponseEntity<Page<Person>> getPerson(Pageable pageable, @RequestParam(defaultValue="", required=false) String firstname,
+			@RequestParam(defaultValue="", required=false) String lastname,
 			@RequestParam(defaultValue="0", required=false) Integer role){
 		
-		 if((id==0)&&(role==0)){
+		 if(role==0){
 			 			 
 			 if(!"".equals(firstname)){
-				 return ResponseEntity.ok(personLogic.getPersonByFirstName(firstname));
+				 return ResponseEntity.ok(personLogic.getPersonByFirstName(pageable, firstname));
 				 
-			 } else if (!"".equals(lastname)){
-				 return ResponseEntity.ok(personLogic.getPersonByLastName(lastname));
+			 } else if (!("".equals(lastname))){
+				 return ResponseEntity.ok(personLogic.getPersonByLastName(pageable, lastname));
 				 
 			 } else{
-				 return  ResponseEntity.ok(personLogic.getAllPersons());
+				 return  ResponseEntity.ok(personLogic.getAllPersons(pageable));
 			 }
 			 
-		 } else if (id==0 && role!=0){
+		 } else {
 			 if(role==1){
 				
-					return ResponseEntity.ok(personLogic.getAllTrainees());
+					return ResponseEntity.ok(personLogic.getAllTrainees(pageable));
 					
 					} else if (role == 2) {
-						return ResponseEntity.ok(personLogic.getAllTrainers());
+						return ResponseEntity.ok(personLogic.getAllTrainers(pageable));
 						
 					} else {
 						return ResponseEntity.ok(null);
 					}
 		 }
 		 
-		 // search for person by id regardless if entered role is incorrect
-		 else{
-			 
-			 List<Person> singlePerson = new ArrayList<>();
-			 
-			 singlePerson.add(personLogic.getPersonById(id));
-			
-			 return ResponseEntity.ok(singlePerson);
-			 
-		 }
 	}
 	
+	
 	@RequestMapping(method = RequestMethod.POST, value = "persons")
-	public ResponseEntity<Person> createPerson(@RequestParam(required=true) String firstname,
-			@RequestParam(required=true) String lastname, 
-			@RequestParam(required=true) Integer role,
-			@RequestParam(defaultValue="0", required=false) Integer batchId){
-			
-			Batch batch = batchLogic.getBatchById(batchId);
-			
-			PersonRole pRole = personRoleLogic.findRoleById(role);
-			Person person = new Person(firstname, lastname, pRole);
-			
-			personLogic.createPerson(person, batch);
-
-			return ResponseEntity.ok(person);
-		
+	public ResponseEntity<Person> createPerson(@RequestBody Person newPerson){
+		System.out.println(newPerson);	
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Replace this usage of System.out or System.err by a logger. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.revaturelabs.com/coding_rules#rule_key=squid%3AS106)